data_validation/__main__.py

```diff
@@ -175,24 +187,42 @@ def build_config_from_args(args, config_manager):
         target_query_str = config_manager.get_query_from_file(args.target_query_file)
         config_manager.append_target_query(target_query_str)
 
+        # For custom-query column command
         if args.custom_query_type == consts.COLUMN_VALIDATION.lower():
             config_manager.append_aggregates(get_aggregate_config(args, config_manager))
 
-        elif not args.primary_keys:
-            raise ValueError(
-                "Expected valid primary keys for custom query row validation, got None."
+        # For custom-query row command
+        if args.custom_query_type == consts.ROW_VALIDATION.lower():
+
+            # Append calculated fields: --hash/--concat
+            config_manager.append_calculated_fields(
```

Comment:
We should keep the append_calculated_fields() outside of the `if config_manager.validation_type == consts.CUSTOM_QUERY` block since calculated fields can be applied to column validations as well, i.e doing a cast before a SUM.

Reply:
In the definition of `get_calculated_config(args, config_manager)`, `calculated_configs` is only populated in case when `--hash` or `--concat` are present. So it returns an empty list in case of `column validation`